src/devices/realsense2/realsense2withIMUDriver.cpp

```diff
@@ -0,0 +1,495 @@
+/*
+ * Copyright (C) 2006-2020 Istituto Italiano di Tecnologia (IIT)
+ * All rights reserved.
+ *
+ * This software may be modified and distributed under the terms of the
+ * BSD-3-Clause license. See the accompanying LICENSE file for details.
+ */
+
+#include "realsense2withIMUdriver.h"
+
+#include <yarp/os/Value.h>
+
+#include <yarp/sig/ImageUtils.h>
+
+#include <yarp/math/quaternion.h>
+#include <yarp/math/math.h>
+
+#include <algorithm>
+#include <cmath>
+#include <cstdint>
+#include <iomanip>
+#include <librealsense2/rsutil.h>
+#include <librealsense2/rs.hpp>
+#include <mutex>
+
+ /**********************************************************************************************************/
+ // This software module is experimental.
+ // It is provided with uncomplete documentation and it may be modified/renamed/removed without any notice.
+ /**********************************************************************************************************/
+ 
+using namespace yarp::dev;
+using namespace yarp::sig;
+using namespace yarp::os;
+
+using namespace std;
+
+#ifndef M_PI
+#define M_PI 3.14159265358979323846
+#endif
+
+struct float3 {
+    float x, y, z;
+    float3 operator*(float t)
+    {
+        return { x * t, y * t, z * t };
+    }
+
+    float3 operator-(float t)
+    {
+        return { x - t, y - t, z - t };
+    }
+
+    void operator*=(float t)
+    {
+        x = x * t;
+        y = y * t;
+        z = z * t;
+    }
+
+    void operator=(float3 other)
+    {
+        x = other.x;
+        y = other.y;
+        z = other.z;
+    }
+
+    void add(float t1, float t2, float t3)
+    {
+        x += t1;
+        y += t2;
+        z += t3;
+    }
+};
+
+class rotation_estimator

```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The class 'rotation_estimator' does not have a constructor although it has private member variables.](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=4772294)